File: LeetCode/Easy/1822 - Sign of the Product of an Array/1822 - Sign of the Product of an Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var arraySign = function(nums) {
    let product = 1;
    
    for (let num of nums) product *= num;
    
    return (product ? (product > 0 ? 1 : -1) : 0);
};

File: LeetCode/Easy/1822 - Sign of the Product of an Array/1822 - Sign of the Product of an Array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int ret = 1;
        for (auto num : nums) {
            if (!num) return 0;
            ret *= (num > 0 ? 1 : -1);
        }
        return ret;
    }
};